### PR TITLE
use newer http-client

### PR DESCRIPTION
**Pull request by bryanmacfarlane**
bump to newer http client

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -47,8 +47,9 @@ jobs:
     - name: Lint
       run: npm run lint
 
-    - name: Format
-      run: npm run format-check
+    # uncomment after reviewing rules
```
Comment (thboop) [inline .github/workflows/unit-tests.yml:50]: Why do we need these changes?

Comment (bryanmacfarlane) [reply .github/workflows/unit-tests.yml:50]: I had commented these out as part of prettier changes and forgot to uncomment as part of last checkin.  Unrelated to version bump

Comment (thboop) [reply .github/workflows/unit-tests.yml:50]: Lets remove these changes for this PR then!

File: packages/github/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@actions/github",
-  "version": "2.1.1",
+  "version": "2.1.2",
```
Comment (thboop) [inline packages/github/package.json:3]: We should update `Releases.MD` for all of these packages as well

File: packages/artifact/package-lock.json [hunk 3]
```diff
@@ -4,11 +4,6 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
-    "@actions/core": {
-      "version": "1.2.3",
-      "resolved": "https://registry.npmjs.org/@actions/core/-/core-1.2.3.tgz",
-      "integrity": "sha512-Wp4xnyokakM45Uuj4WLUxdsa8fJjKVl1fDTsPbTEcTcuu0Nb26IPQbOtjmnfaCPGcaoPOOqId8H9NapZ8gii4w=="
-    },
     "@actions/http-client": {
       "version": "1.0.7",
```
Comment (thboop) [inline packages/artifact/package-lock.json:8]: this didn't update

File: packages/artifact/package-lock.json [hunk 4]
```diff
@@ -4,11 +4,6 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
-    "@actions/core": {
```
Comment (thboop) [inline packages/artifact/package-lock.json:7]: Note regarding failing tests here: https://github.com/actions/toolkit/pull/421/files#r414150872

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (bryanmacfarlane) [reply .github/workflows/unit-tests.yml:50]: sure. will do.

File: packages/github/package.json (hunk 2 above)
Comment (bryanmacfarlane) [reply packages/github/package.json:3]: Yep.  Per IM, I said I was going to finish this PR tomorrow.  I converted to a draft to make that clearer.

**Comment by bryanmacfarlane** [comment]
I'm going to abandon this PR and just change the tool-cache (the only thing affected and needing the http client update

**Pull request closed without merging**